pkg/api/handlers/libpod/images.go

```diff
@@ -38,17 +38,22 @@ func ImageExists(w http.ResponseWriter, r *http.Request) {
 }
 
 func ImageTree(w http.ResponseWriter, r *http.Request) {
-	// tree is a bit of a mess ... logic is in adapter and therefore not callable from here. needs rework
-
-	// name := utils.GetName(r)
-	// _, layerInfoMap, _, err := s.Runtime.Tree(name)
-	// if err != nil {
-	//	Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrapf(err, "Failed to find image information for %q", name))
-	//	return
-	// }
-	//	it is not clear to me how to deal with this given all the processing of the image
-	// is in main.  we need to discuss how that really should be and return something useful.
-	handlers.UnsupportedHandler(w, r)
+	runtime := r.Context().Value("runtime").(*libpod.Runtime)
+	name := utils.GetName(r)
+
+	img, err := runtime.ImageRuntime().NewFromLocal(name)
+	if err != nil {
+		utils.Error(w, "Something went wrong.", http.StatusNotFound, errors.Wrapf(err, "Failed to find image %s", name))
+		return
+	}
+
+	tree, err := img.NewTree(true)
```

Comment:
We need a whatrequires paramater here as well.

Reply:
Added the parameter as well.